plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java

```diff
@@ -3859,4 +3878,33 @@ public boolean supportsReportingWrittenBytes(ConnectorSession session, SchemaTab
     {
         return true;
     }
+
+    public static class TableParameterProvider
+    {
+        private final Table table;
+        private final Set<String> consumedParameters = new HashSet<>(TABLE_PROPERTIES_TO_SKIP);
+
+        public TableParameterProvider(Table table)
+        {
+            this.table = requireNonNull(table, "table is null");
+        }
+
+        public String getParameter(String key)
+        {
+            consumedParameters.add(key);
+            return table.getParameters().get(key);
+        }
+
+        public Table getTable()
+        {
+            return table;
+        }
+
+        public Map<String, String> getUnconsumedProperty()
+        {
+            return table.getParameters().entrySet().stream()
```

Comment:
This could be tracked incrementally as part of `getParameter(String key)` by taking an initial snapshot of `table.getProperties()` in the constructor, removing each property that gets consumed, and just returning the remaining map entries at this point.

Reply:
But if we need to use the same properties at multiple places then we might not be able to handle them.